src/components/NewTodo/NewTodo.tsx

```diff
@@ -1,47 +1,53 @@
 import { Button, Grid, Stack, TextField } from "@mui/material";
-import { useRef } from "react";
+import { useForm } from "react-hook-form";
+import { useMutation } from "react-query";
 
-const NewTodo = ({ onAddTodo }) => {
-  const titleRef = useRef<HTMLInputElement>(null);
-  const textRef = useRef<HTMLInputElement>(null);
+import { instance } from "../../../config";
+import Todo from "../../models/todo";
 
-  const submitHandler = (event) => {
-    event.preventDefault();
+const NewTodo = () => {
+  const { register, handleSubmit } = useForm({
+    defaultValues: { title: "", text: "" },
+  });
 
-    const enteredTitle = titleRef.current.value;
-    const enteredText = textRef.current!.value;
+  const addTodoHandler = (todoData: Todo) => {
+    const response = instance.post("/todos", {
+      title: todoData.title,
+      text: todoData.text,
+    });
 
-    const todoData = {
-      title: enteredTitle,
-      text: enteredText,
-    };
+    return response;
+  };
+
+  const { mutate } = useMutation(addTodoHandler);
 
-    onAddTodo(todoData);
+  const submitHandler = (data: Todo) => {
+    mutate(data);
   };
 
   return (
-    <form onSubmit={submitHandler}>
+    <form onSubmit={handleSubmit(submitHandler)}>
```

Comment:
No need to use the `submitHandler` also. Just use the `mutate` function

Reply:
couldn't get it working with only mutate